packages/compiler-cli/src/ngtsc/typecheck/src/template_symbol_builder.ts

```diff
@@ -150,7 +150,24 @@ export class SymbolBuilder {
   private getDirectiveMeta(
       host: TmplAstTemplate|TmplAstElement,
       directiveDeclaration: ts.Declaration): TypeCheckableDirectiveMeta|null {
-    const directives = this.templateData.boundTarget.getDirectivesOfNode(host);
+    let directives = this.templateData.boundTarget.getDirectivesOfNode(host);
+
+    // `getDirectivesOfNode` will not return the directives intended for an element
+    // on a microsyntax template, for example `<div *ngFor="let user of users;" dir>`,
+    // the `dir` will be skipped, but it's needed in language service.
+    const firstChild = host.children[0];
+    if (firstChild instanceof TmplAstElement) {
+      const isMicrosyntaxTemplate = host instanceof TmplAstTemplate &&
+          sourceSpanEqual(firstChild.sourceSpan, host.sourceSpan);
+      if (isMicrosyntaxTemplate) {
```

Comment:
```suggestion
      const isMicrosyntaxTemplate = firstChild instanceof TmplAstElement && host instanceof TmplAstTemplate &&
          sourceSpanEqual(firstChild.sourceSpan, host.sourceSpan);
      if (isMicrosyntaxTemplate) {
```
nit: let's just combine all of these into one variable for `isMicrosyntaxTemplate`, which is truly a combination of them all 3 of these.

Reply:
If combine these into one variable, the ts cannot narrow in branches guarded by `if (isMicrosyntaxTemplate)`
Or remove the variable:
```
if (firstChild instanceof TmplAstElement && host instanceof TmplAstTemplate &&
        sourceSpanEqual(firstChild.sourceSpan, host.sourceSpan)) {
}
```

Reply:
👍 Ah, I see. I like the variable as a sort of documentation by variable name.